File: Week 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;

    char punctuation[3] = {'.', '!', '?'};

    string checker = get_string("Text: ");
    int length = strlen(checker);

    for (int i = 0; i < length; i++)
    {
        if (isalpha(checker[i]))
        {
            letters++;
        }

        for (int k = 0; k < 3; k++)
        {
            if (checker[i] == punctuation[k])
            {
                sentences++;
            }
        }

        if (checker[i] == ' ')
        {
            words++;
        }
    }

    double L = (letters / (double) words) * 100;
    double S = (sentences / (double) words) * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        int findex = round(index);
        printf("Grade %d\n", findex);
    }
}

File: cs50x - Introduction to Computer Science/week6-python/readability.py
from cs50 import get_string
import re


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)


    index = calculate_coleman_liau_index(letters, words, sentences)


    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def count_letters(text):

    return sum(1 for char in text if char.isalpha())


def count_words(text):

    return len(text.split())


def count_sentences(text):

    return sum(1 for char in text if char in ".!?")


def calculate_coleman_liau_index(letters, words, sentences):

    L = (letters / words) * 100
    S = (sentences / words) * 100

    return round(0.0588 * L - 0.296 * S - 15.8)


if __name__ == "__main__":
    main()
